advanced-achievements-plugin/src/main/java/com/hm/achievement/command/completer/CommandTabCompleter.java

```diff
@@ -88,15 +99,30 @@ public void extractConfigurationParameters() {
 	/**
 	 * Returns a partial list based on the input set. Members of the returned list must start with what the player has
 	 * types so far. The list also has a limited length to avoid filling the player's screen.
-	 * 
+	 *
+	 *
+	 * @param sender
 	 * @param fullSet
 	 * @param prefix
 	 * @return a list limited in length, containing elements matching the prefix,
 	 */
-	private List<String> getPartialList(Set<String> fullSet, String prefix) {
+	private List<String> getPartialList(CommandSender sender, Set<String> fullSet, String prefix) {
+		if (sender instanceof ConsoleCommandSender) {
+			// Console mapper uses the given parameters, spaces and all.
+			return getFormattedPartialList(fullSet, prefix, Function.identity());
+		} else {
+			// Default mapper replaces spaces with an Open Box character to prevent completing wrong word.
+			// Prevented Behaviour:
+			// T -> Tamer -> Teleport Man -> Teleport The Avener -> Teleport The The Smelter
+			return getFormattedPartialList(fullSet, prefix, s -> StringUtils.replace(s, " ", "\u2423"));
```

Comment:
Why I didn't replace the Open box character anywhere else is because the current implementation of other commands that use tab completion do not perform any replacement either
(Ie. [parseAchievementName in AbstractParsableCommand](https://github.com/PyvesB/AdvancedAchievements/blob/master/advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/AbstractParsableCommand.java#L64) 

Difficulty in performing this is spitting each argument by the character and turning the argument arrays into a new array.
I think this should be performed inside `AbstractCommand` `execute` method so that other commands also work with the hack in place, or in `PluginCommandExecutor ` since that is first point of entry for the arguments. Which would you prefer?

Reply:
I'm slightly confused about your first statement - the is reverse replacement on [this line](https://github.com/PyvesB/AdvancedAchievements/blob/5603893f8e91cc593c67c94162e85c2e3e857b89/advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/AbstractParsableCommand.java#L73).

But I agree we should go for a more encompassing approach. I'm okay with performing the arguments pre-processing in _PluginCommandExecutor.java_. This class has some nice unit tests, let's make sure we cover the new open box stuff as well. 😉

Reply:
Ohh, I completely missed that line.
I'm guessing I misattributed the Open box characters not working on inspect command (For now obivious reasons) to other commands not working in my mind somehow.

I'll add that pre-processing to _PluginCommandExecutor.java_ and tests

Reply:
Implemented and tested. :)